frontend/cypress/support/commands.ts

```diff
@@ -24,49 +24,58 @@
+     * @example cy.getBySel('greeting')
+     */
+    getBySel(selector: string, ...args: any): Chainable<Subject>;
 
-declare global {
-	namespace Cypress {
-		interface Chainable<Subject> {
-			login(providerName?: string, username?: string, password?: string, auth_strategy?: string): Chainable<Element>;
-			logout(): Chainable<Element>;
-		}
-	}
+    login(providerName?: string, username?: string, password?: string, auth_strategy?: string): Chainable<Subject>;
+
+    logout(): Chainable<Subject>;
+  }
 }
 
+let haveCookie = Cypress.env('cookie');
+
 Cypress.Commands.add('login', (provider: string, username: string, password: string, auth_strategy: string) => {
-	cy.log("auth cookie is:", haveCookie)
+  cy.log('auth cookie is:', haveCookie);
 
-	cy.window().then((win: any) => {
-		if (auth_strategy != 'openshift') {
-			cy.log('Skipping login, Kiali is running with auth disabled');
-		} else {
-			if (haveCookie === false) {
-				cy.intercept('api/authenticate').as('authorized') //request setting kiali cookie
-				cy.log(
-					`provider: ${provider}, 
+  cy.window().then((win: any) => {
+    if (auth_strategy != 'openshift') {
+      cy.log('Skipping login, Kiali is running with auth disabled');
+    } else {
+      if (haveCookie === false) {
+        cy.intercept('api/authenticate').as('authorized'); //request setting kiali cookie
+        cy.log(
+          `provider: ${provider}, 
 						username: ${username},
-						atuhorization: ${auth_strategy}`)
-				cy.visit('/');
+						atuhorization: ${auth_strategy}`
+        );
+        cy.visit('/');
 
-				if (auth_strategy === 'openshift') {
-					cy.get('.pf-c-form').contains(provider, { matchCase: false }).click()
-				}
+        if (auth_strategy === 'openshift') {
+          cy.get('.pf-c-form').contains(provider, { matchCase: false }).click();
+        }
 
-				if (auth_strategy === 'openshift') {
-					cy.get('#inputUsername').type(username);
-					cy.get('#inputPassword').type(password);
-					cy.get('button[type="submit"]').click()
-					cy.wait('@authorized').its('response.statusCode').should('eq', 200)
-					cy.getCookie('kiali-token', { timeout: 5000 }).should('exist').then((c) => {
-						haveCookie = true
-					})
-				}
-			} else {
-				cy.log('got an auth cookie, skipping login')
-			}
-		}
-	});
+        if (auth_strategy === 'openshift') {
+          cy.get('#inputUsername').type(username);
+          cy.get('#inputPassword').type(password);
+          cy.get('button[type="submit"]').click();
+          cy.wait('@authorized').its('response.statusCode').should('eq', 200);
+          cy.getCookie('kiali-token', { timeout: 5000 })
+            .should('exist')
+            .then(c => {
+              haveCookie = true;
+            });
+        }
+      } else {
+        cy.log('got an auth cookie, skipping login');
+      }
+    }
+  });
 });
 
+Cypress.Commands.add('getBySel', (selector: string, ...args) => cy.get(`[data-cy="${selector}"]`, ...args));
```

Comment:
I'm not sure if this helper would be seen as a "magic" function.
As commented, and it's shown in the code of this PR
![image](https://user-images.githubusercontent.com/1662329/164196138-288e486f-479e-4e3e-90e9-f4fa9bbae29c.png)

I'd prefer to see the real selector there, as I think there may be cases where the attribute may change.

Reply:
Yes including it here add its to the `cy` interface so you can query/chain it the same way you would other `cy.` commands. It's basically the same thing as:
https://github.com/cypress-io/cypress-realworld-app/blob/develop/cypress/support/commands.ts#L34-L36
https://github.com/openshift/console/blob/master/frontend/packages/integration-tests-cypress/support/selectors.ts#L36-L41
which make it more convenient to query for `cy.get(["data-test-unique-selector"])`. Even if we have additional attributes, it'd still be nice to have some syntactic sugar around the base case of `cy.get(["data-test-${selector}"])`.